tox.ini

```diff
@@ -1,48 +1,167 @@
+    future:   {[future-config]basepython}
+
+deps=
+    current:  {[current-config]deps}
+    future:   {[future-config]deps}
+    lint:     {[lint-config]deps}
+    unittest: {[unittest-config]deps}
+
+passenv=
+    unittest:   {[unittest-config]passenv}
+
+setenv=
+    unittest: {[unittest-config]setenv}
+
+commands=
+    lint:     {[lint-config]commands}
+    unittest: {[unittest-config]commands}
+
+
+[lint-config]
+# Configuration necessary to lint Python files.
+# Note: This is not an env and will not run if invoked. Use:
+#
+#   tox -e lint-{current, future}
+#
+# To run Python linting.
+deps=
+    black
+    flake8
+    isort
+commands=
+    black --check teachers_digital_platform setup.py
+    flake8 teachers_digital_platform
+    isort --check-only --diff --recursive teachers_digital_platform
+
+
+[unittest-config]
+# Configuration necessary to run unittests.
+# Note: This is not an env and will not run if invoked. Use:
+#
+#   tox -e unittest-{current, future}
+#
+# To run unit tests.
 install_command=pip install -e ".[testing]" -U {opts} {packages}
+passenv=
+    TRAVIS TRAVIS_JOB_ID TRAVIS_BRANCH TEST_DATABASE_URL TEST_RUNNER
+setenv=
+    DJANGO_SETTINGS_MODULE=teachers_digital_platform.tests.settings
+    PYTHONPATH={toxinidir}/cfgov-refresh/cfgov:{env:PYTHONPATH:}
+    TEST_DATABASE_URL=postgres://cfpb@localhost/cfgov
+deps=
+    current: {[current-config]deps}
 commands=
-    lint: {[lint]commands}
     ./install_cfgov_refresh.sh
     coverage erase
     coverage run --source='teachers_digital_platform' {envbindir}/django-admin.py test {posargs}
     coverage report -m
-setenv=
-    DJANGO_SETTINGS_MODULE=teachers_digital_platform.tests.settings
-    PYTHONPATH={toxinidir}/cfgov-refresh/cfgov:{env:PYTHONPATH:}
 
-basepython=
-    py36: python3.6
 
+[current-config]
+# Configuration necessary for the production versions Python, Django and 
+# Wagtail
+# Note: This is not an env and will not run if invoked. Use:
+# 
+#   tox -e unittest-current
+#   tox -e unittest
+# 
+# To run unittests with the current production versions.
+basepython=python3.6
 deps=
-    dj111: Django>=1.11,<1.12
-    dj20: Django>=2.0,<2.1
-    dj22: Django>=2.2,<2.3
-    wag23: wagtail>=2.3,<2.4
-    wag28: wagtail>=2.8,<2.9
-    lint:  {[lint]deps}
-
-[lint]
+    Django>=1.11,<1.12
+    wagtail>=2.3,<2.4
+
+
+[future-config]
+# Configuration necessary for upcoming versions of Python, Django, and 
+# Wagtail. This is the place where the "future" dependencies will need to be 
+# modified when we're ready to track new versions of Python, Django, or 
+# Wagtail.
+# Note: This is not an env and will not run if invoked. Use:
+# 
+#   tox -e unittest-future
+# 
+# To run unittests with the upcoming versions specified here.
+basepython=python3.6
 deps=
-    flake8>=2.2.0
-    isort>=4.2.15
-commands=
-    flake8 .
-    isort --check-only --diff --recursive teachers_digital_platform
+    Django>=2.0,<2.1
+    wagtail>=2.8,<2.9
```

Comment:
I actually wonder if there's a way we could centralize the version specification of `future`, so that they're only ever defined in one place, in cfgov-refresh, if we're going to go this route of not including them as factors in the tox environment names.

Reply:
There is a circular dependency between TDP and cfgov-refresh.
TDP has `install_cfgov_refresh.sh` which has the lines:
```
cd cfgov-refresh
pip install $(
    cat requirements/libraries.txt | \
    awk '!/teachers_digital_platform/ && !/^#/'
)
pip install -r requirements/test.txt
```
TDP uses the versions of Django and wagtail defined in `requirements/libraries.txt`

Reply:
Yeah, I don't know if there's a clean way to do it.